visualization_scripts/run.py

```diff
@@ -0,0 +1,175 @@
+""" 
+script loads style file and TPS output file into Cytoscape session
+
+@author: Adam Shedivy
+
+"""
+
+# imports 
+from py2cytoscape.data.cynetwork import CyNetwork
+from py2cytoscape.data.cyrest_client import CyRestClient
+from py2cytoscape.data.style import StyleUtil
+import py2cytoscape.util.cytoscapejs as cyjs
+import py2cytoscape.cytoscapejs as renderer
+from py2cytoscape import cyrest
+import networkx as nx
+import pandas as pd
+import json
+import sys
+import time
+import subprocess
+import os
+from requests.exceptions import ConnectionError as CE
+from json.decoder import JSONDecodeError 
+
+#assert Cytoscape is running on machine
+
+def process_exists(process_name):
+    """
+    function to assert Cytoscape is running
+
+    Args:
+        process_name (str): name of process to check for
+        example - 'Cytoscape.exe'
+
+    Returns:
+        true if process runnning, else false
+
+    """
+    bytes_name = str.encode(process_name)
+    call = 'TASKLIST', '/FI', 'imagename eq %s' % process_name
+
+    # use buildin check_output right away
+    output = subprocess.check_output(call)
+
+    # check in last line for process name
+    last_line = output.strip().split(b'\r\n')[-1]
+
+    return last_line.lower().startswith(bytes_name.lower())
+
+
+def find_path(name, path):
+    """
+    Search Algorithm to find Cytoscape application on machine
+
+    Args:
+        name (str): name of application 
+        path (str): top directory to start search from 
+
+    Returns:
+        path to first match on machine
+    """
+
+    # list to hold found matches
+    result = []
+
+    # walk directory/file tree 
+    for root, dirs, files in os.walk(path):
+        if name in files:
+            result.append(os.path.join(root, name))
+    return result[0]
+
+
+def vis(output, style):
+    """
+    REQUIRED ARGS:
+        output: output file from tps software
+        style: style file for cytoscpe 
+    """
+    
+    #create cyrest client
+    cyy = cyrest.cyclient()
+    
+    # Step 1: Create py2cytoscape client
+    cy = CyRestClient()
+
+    # Reset
+    cy.session.delete()
+    print("---session reset")
+
+    # Step 2: Load network from somewhere
+    print("---loading network file")
+    tps_net = cy.network.create_from(output)
+    print("---Done")
+    time.sleep(2)

```

Comment:
Is the sleep command needed here?

Reply:
This command comes in handy if a Cytoscape session is already open, and you want to load a new style file on the same graph data. Without pausing the program, the Client is unsuccessful in loading the new style file and the output file is loaded as a default style. Through trial and error, I found that the Cytoscape client only needs a couple of seconds to fully refresh. You can try to test this on your machine and see if you run into the same issues.

To recreate: run the script as usual. Then using the style file that was provided for the manuscript (tps_style.xml) https://github.com/agitter/tps/blob/examples/output/tps_style.xml,
run the script again but leave the Cytoscape application open. See if the new style loaded correctly, comment out the sleep command and try again to see if the styles are loading correctly.